### PR TITLE
propose HIPE: Agent message routing

### PR DESCRIPTION
**Pull request by tplooker**
As discussed in the Agent WG call on Dec 5, this HIPE defines the following 

- New message types for the routing family to maintain routing records.
- Two examples of message delivery featuring routing and maintenance of routing records.

Please pay close attention to some of the un-resolved questions around this implementation as I am not convinced with all aspects.

### CODE REVIEW

File: text/agent-message-routing/README.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+In the presence of this connection, Bob's agent prepares the following message to agents-r-us.
+
+```json
+{
+ "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routing/0.1/create",
+ "recipient-identifier" : "<B.did@B:A>"
+}
+```
+
+Note - with the wire level message used to transport the above, agents-r-us MUST be able to recover the sender. As this is the basis for the routing record.
```
[the remark this answers is not in the document]
Comment (tplooker) [reply text/agent-message-routing/README.md]: Interesting, im not sure about that, correct me if Im wrong but you are saying perhaps state the endpoint for the next hop in the routing message? In that implementation whats to stop me giving someone else's endpoint in the pursuit of spamming them? 

By enforcing that create routing record messages must be authpacked, the sender can be validated and linked to a pre-existing connection, preventing the need to re validate an endpoint. Essentially in the implementation I've stated we are leaning on the trusted connection already established between the recipient and the router.

I tried to raise this point a while ago in regards to are we going to constrain some messages in that they must be authpack'd in order to be valid?

**Comment by dhh1128** [comment]
@tplooker How do you want to move this into Aries, if at all?

**Comment by tplooker** [comment]
@dhh1128 yes I am working on a revision to this HIPE

**Comment by dhh1128** [comment]
I believe this PR is superseded, both by Aries work and by DIDComm work at DIF. Closing. If I'm wrong, please reopen.

**Pull request closed without merging**